shared/naturalcrit/markdown.js

```diff
@@ -238,7 +268,38 @@ const definitionLists = {
 	}
 };
 
-Marked.use({ extensions: [mustacheSpans, mustacheDivs, mustacheInjectInline, definitionLists] });
+let pageBlockTopLevel = true;
+let pageBlockNumber = 0;
+const pageBlocks = {
+	name  : 'pageBlock',
+	level : 'block',
+	start(src) { return pageBlockTopLevel; },
```

Comment:
`start:` isn't supposed to return a boolean. It returns the index of where the token starts.

Since Marked scans for tokens in the order the extensions are loaded, it's possible for it to see one token and jump ahead in the string, leaving later tokens unprocessed. `start` places a marker at any likely places so it knows to stop and check before jumping ahead.

For example, if we had:

```
{{curlystuff

DL :: STUFF

}}
```

And we had loaded the DL extension first, Marked would jump past the curlies and not process them because it was looking for DL first.

With `start` filled in for both (using a simplified regex that just looks for `{{`), it knows to make sure to stop at the curlies and check the other extensions before moving on.

Reply:
In our case, page blocks should always be assumed to start at the very beginning of the current string, so we should have `start: 0` or just no `start` at all.

Having `start: true` gets converted to 1, so it's adding a marker 1 char after the start. Extensions are always checked at the beginning of the string too, so the first time the page extension comes around it already eats up the whole page, and that marker will get discarded anyway.

Reply:
As per other comments, `start(src)` has been removed as it was no longer doing anything.